modules/assim.sequential/R/downscale_function.R

```diff
@@ -3,36 +3,32 @@
 ##' @author Joshua Ploshay
 ##'
 ##' @param data  In quotes, file path for .rds containing ensemble data.
-##' @param focus_year In quotes, if SDA site run, format is yyyy/mm/dd, if NEON, yyyy-mm-dd. Restricted to years within file supplied to 'data'.
+##' @param coords In quotes, file path for .csv file containing the site coordinates, columns named "lon" and "lat".
+##' @param date In quotes, if SDA site run, format is yyyy/mm/dd, if NEON, yyyy-mm-dd. Restricted to years within file supplied to 'data'.
 ##' @param C_pool In quotes, carbon pool of interest. Name must match carbon pool name found within file supplied to 'data'.
-##' @param covariates In quotes, file path of SpatRaster stack, used as predictors in randomForest. Layers within stack should be named.
-##' @param cords In quotes, file path for .csv file containing the site coordinates, columns named "lon" and "lat".
+##' @param covariates Should be loaded in using 'covariates' instructions in inst folder
 ##' @details This function will downscale forecast data to unmodeled locations using covariates and site locations
 ##'
 ##' @description This function uses the randomForest model.
 ##'
 ##' @return It returns the `downscale_output` list containing lists for the training and testing data sets, models, and predicted maps for each ensemble member.
 
 
-NA_downscale <- function(data, cords, covariates, focus_year, C_pool){
-  
-  # Read in the covariates and set CRS to EPSG:4326
-  covariates <- terra::rast(covariates) # ADD package to every function
-  terra::crs(covariates) <- "EPSG:4326"
+NA_downscale <- function(data, coords, date, C_pool){
   
   # Read the input data and site coordinates
   input_data <- readRDS(data)
-  site_coordinates <- terra::vect(readr::read_csv(cords), geom=c("lon", "lat"), crs="EPSG:4326")
+  site_coordinates <- terra::vect(readr::read_csv(coords), geom=c("lon", "lat"), crs="EPSG:4326")
   
   # Extract the carbon data for the specified focus year
-  index <- which(names(input_data) == focus_year)
+  index <- which(names(input_data) == date)
   data <- input_data[[index]]
   carbon_data <- as.data.frame(t(data[which(names(data) == C_pool)]))
   names(carbon_data) <- paste0("ensemble",seq(1:ncol(carbon_data)))
   
   # Extract predictors from covariates raster using site coordinates
   predictors <- as.data.frame(terra::extract(covariates, site_coordinates)) 
-  predictors <- dplyr::select(predictors, -1)
+  predictors <- dplyr::select(predictors, -ID)
```

Comment:
I don't see where `ID` is being defined/calculated in the code above.

Reply:
Looks like it was added by `terra::extract` on line 34 above, and could be suppressed by by passing `ID = FALSE` there